### ISSUE TITLE
Make Curriculum Printable

### ISSUE THREAD

**Issue by LauraHilliger** [mozilla/webmaker-curriculum#99]
Printable curriculum is widely requested (as is tablet/mobile optimized) Is there a way we can write a little thing that makes printing our curriculum super easy? 

If I have this URL:

http://mozilla.github.io/webmaker-curriculum/WebLiteracyBasics-I/session01-credibility.html

and click "print", can we automagically print the activities it links to as well? What about CSS for styling prints? 

cc/ @toolness @iamjessklein

**Comment by toolness** [comment]
I have found it very annoyingly hard to make professional-looking PDF/printable versions of HTML pages. This is partly because there is a whole subset of CSS specific to print (for specifying things like page breaks) that has very little browser support. There is a program called [Prince](http://www.princexml.com/) that is specifically made to support all that print CSS for HTML->PDF generation, but it costs lots of money... It says that it is free for non-profits but that version appears to have an annoying watermark on the output PDFs.

That's basically the extent of my knowledge on this topic, alas. :cry:

**Comment by jgmac1106** [comment]
TLDR: Ink is expensive. Keep it simple

Maybe you shouldn't try to make your paper design look like you CSS Design. People who print worry about ink, especially in developing nations or public schools. 

Make it as plain text as possible. 

Couldn't you just quickly make a template.

Goal:
Competencies
Objectives:
Tools:
Learning Activities
  Title
  Description
  Time
Reflect and review

Maybe some smart dev could figure out a way to scrape the content from the curriculum but sweat equity is probably the best tool.

**Comment by toolness** [comment]
Hmm, good point about saving ink--if it doesn't need to look like a super-slick facsimile of the webpage, then going with a really simple CSS print stylesheet could work really well!

**Comment by dajbelshaw** [comment]
If it's going to be too much hassle, we could just direct people towards sites like http://www.printfriendly.com

**Comment by LauraHilliger** [comment]
@andylolz said on Twitter that Code Club does server side generation using phantom.js

So seems like we have two options - make it easy with a click of a button or create resources that point to an appropriate service.

**Comment by toolness** [comment]
Er, if we just provide a good CSS print stylesheet that minimizes ink usage and maintains decent readability--but doesn't look ultra professional--then we can just tell them to use their browser's "print this page" feature, right?

That's essentially all phantomjs is doing--it's just applying the print stylesheet and printing to a PDF. Granted, there are use-cases where PDF files are better than a browser's "print" feature, e.g. sending the PDF to a tablet device for offline reading, but I'm assuming that the primary value here comes from people actually printing to hardcopy. (Also, for the offline use case we can use appcache, but that's kind of a different story. And if the user is on a Mac, they can easily use the browser's print feature to print to PDF themselves, if they really want one.)

**Comment by toolness** [comment]
Er, I suppose I should also mention that different browsers do print differently, so there is value in using PhantomJS to always print a consistently decent-looking PDF. But we could also just recommend a specific browser for printing.

**Comment by dajbelshaw** [comment]
I'd love to see statistics of people requesting a print function versus actual usage.

Another way to slice-and-dice this would be to version the resources, and create a PDF alternative. The understanding would be that the PDF is 'best effort' and may (for example) be at v1.0 while the web version is at v1.2. Just a thought.

**Comment by jgmac1106** [comment]
I don't know the techno mumbo jumbo. Just teachers. Make it like a recipe
website. It looks all complicated on the web but when you hit the print
icon you get a nice clean sheet with nothing but directions.

**Comment by toolness** [comment]
> I don't know the techno mumbo jumbo. Just teachers. Make it like a recipe website. It looks all complicated on the web but when you hit the print icon you get a nice clean sheet with nothing but directions.

Yep, that is precisely just a well-made print CSS stylesheet. There's a good Smashing guide on [How To Set Up A Print Style Sheet](http://www.smashingmagazine.com/2011/11/24/how-to-set-up-a-print-style-sheet/) from 2011... If you need help with this, I can assist.

**Comment by iamjessklein** [comment]
+1 @jgmac1106 -  plus many of the most useful supplemental materials are printable - I have them in [like every kit](https://jess.makes.org/thimble/MTEzMjMzMTI2NA==/actionable-feedback) I have ever made for Webmaker.

**Comment by dajbelshaw** [comment]
Wondering with @LauraHilliger being away whether @iamjessklein and @toolness could collaborate on a solution we can add into the activity template?
